main.js

```diff
@@ -940,9 +940,8 @@ function delObjects(objs, callback) {
 function checkHost(callback) {
     const objectData = objects.getStatus();
     // only main host controller needs to check/fix the host assignments from the instances
-    if (compactGroupController || !objectData.server) {
-        callback && callback();
-        return;
+    if ((compactGroupController || !objectData.server) && config.system.compact) {
```

Comment:
I think this is wrong. We "just"need to remove the "|| !!objectData.server" from the original if and it shpuld be fixed too.

Adding this was "stupid" and made this error happening with a redis db ... file DB should still have worked!